src/serverenvironment.cpp

```diff
@@ -818,6 +838,15 @@ class ABMHandler
 		{
 			const MapNode &n = block->getNodeUnsafe(p0);
 			content_t c = n.getContent();
+			// cache content types as we go
+			if (!block->contents_cached && !block->do_not_cache_contents) {
+				block->contents.insert(c);
+				if (block->contents.size() > 64) {
+					// too many different nodes... don't try to cache
```

Comment:
'Too'

Reply:
It says "too". You mean upper case T?
It's not a full sentence, though.

Reply:
Yes capital 'T' please.